### DISCUSSION TITLE
Persistent site selection across index and element pages

### DISCUSSION THREAD

**Discussion by jannisborgers** [craftcms/cms#11443] [Ideas] [answered]
Our editors are having problems with the site selection dropdown (*site selector* according to the changelog) not persisting across all pages. 

When editors are viewing an index page with `Site A` selected, clicking an entry always opens the entry for `Site A` via a URL parameter. When they are on the entry page and switch from `Site A` to `Site B`, then click the `Save` button at the top right, they are redirected back to the `Site A` index page. 

This might be the expected behavior. Our editors’ expectation is that changing the site will allow them to _stay_ on that site for all following actions, and will only change if they decide to manually do so.

CraftCMS version is 3.7.43

**Comment by brandonkelly** [comment, marked as answer]
This has already been resolved in Craft 4 – see #8920.

**Reply by jannisborgers** [reply]
Thanks Brandon, didn’t know what to search for. This site isn’t upgraded to Craft 4 yet, but anyway: ☑ Resolved

**Reply by brandonkelly** [reply]
Hah yeah I just searched the Craft 4.0.0 release notes for `site`, since I knew we added it :)
